pkg/controllers/metrics/pod/controller.go

```diff
@@ -319,9 +319,11 @@ func (c *Controller) recordPodStartupMetric(pod *corev1.Pod, schedulableTime tim
 				podName:      pod.Name,
 				podNamespace: pod.Namespace,
 			})
-			PodStartupDurationSeconds.Observe(cond.LastTransitionTime.Sub(pod.CreationTimestamp.Time).Seconds(), nil)
-			if !schedulableTime.IsZero() {
-				PodProvisioningStartupDurationSeconds.Observe(cond.LastTransitionTime.Sub(schedulableTime).Seconds(), nil)
+			if ready {
```

Comment:
Can we actually just handle this in one `if ready` and then perform the same logic in an `else if` for `podutils.IsTerminal`?

Reply:
+1 - I don't mind the minimal duplication in this case since it keeps us explicit about what we're doing for each case. It's also worth leaving a comment regarding why we have to differentiate between ready and terminal pods. I don't think the updated last transition time on the status time is obvious unless you've looked into the issue.

Reply:
Done